websocket/websocket.go

```diff
@@ -0,0 +1,152 @@
+package websocket
+
+import (
+	"errors"
+	"log"
+	"net/url"
+	"os"
+	"os/signal"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+// Client for forex api instance
+type Client struct {
+	Address   *string
+	Conn      *websocket.Conn
+	Listeners *[]func(interface{})
```

Comment:
pointer to slice

Reply:
for managing the slice purpose (add, delete)